esrally/track/params.py

```diff
@@ -692,7 +707,21 @@ def __iter__(self):
     def __next__(self):
         if self.conflicting_ids is not None:
             if self.conflict_probability and self.id_up_to > 0 and self.rand() <= self.conflict_probability:
-                doc_id = self.conflicting_ids[self.randint(0, self.id_up_to - 1)]
+                # a recency of zero means that we don't care about recency and just take a random number
+                # within the whole interval.
+                if self.recency == 0:
+                    idx = self.randint(0, self.id_up_to - 1)
+                else:
+                    # A recency > 0 biases id selection towards more recent ids. The recency parameter decides
+                    # by how much we bias. See docs for the resulting curve.
+                    #
+                    # idx_range is in the interval [0;1].
+                    idx_range = min(self.randexp(GenerateActionMetaData.RECENCY_SLOPE * self.recency), 1)
+                    # the resulting index is in the range [0, self.id_up_to). Note that a smaller idx_range
+                    # biases towards more recently used ids (higher indexes).
+                    idx = int(round((self.id_up_to - 1) * (1 - idx_range)))
```

Comment:
Does `round()` anyway return an `int`, if the optional [ndigits parameter](https://docs.python.org/3/library/functions.html#round) is not passed?

No harm being explicit here, but wondering about it.

Reply:
Oh, you are right. I was probably assuming that it would behave similarly to `Math#round`in Java, hence the explicit conversion.